src/ai-providers/codex-cli.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * src/ai-providers/codex-cli.js
+ *
+ * Codex CLI provider implementation using the ai-sdk-provider-codex-cli package.
+ * This provider uses the local OpenAI Codex CLI with OAuth (preferred) or
+ * an optional OPENAI_API_KEY if provided.
+ */
+
+import { createCodexCli } from 'ai-sdk-provider-codex-cli';
+import { BaseAIProvider } from './base-provider.js';
+import { execSync } from 'child_process';
+import { log } from '../../scripts/modules/utils.js';
+import { getCodexCliSettingsForCommand } from '../../scripts/modules/config-manager.js';
+
+let _codexCliChecked = false;
+let _codexCliAvailable = null;
```

Comment:
why not add it as a const inside the class ? that way we can reference it with this._codexCliChecked ?